### PR TITLE
GH-1888: As a smith I want the builder to not reuse EMF Resources, EObjects, etc.

### PR DESCRIPTION
**Pull request by mor-n4**
See #1888.

Maybe there will be a second part for using `unloadAST()`.

### CODE REVIEW

File: plugins/org.eclipse.n4js/src/org/eclipse/n4js/postprocessing/AbstractProcessor.xtend [hunk 1]
```diff
@@ -207,11 +210,13 @@ package abstract class AbstractProcessor {
 		System.out.flush();
 		System.err.println(msg);
 		System.err.flush();
+		LOG.error(msg);
 	}
 
 	def protected static Throwable logException(Throwable th) {
 		// always log exceptions, even if !isDEBUG_LOG()
 		th.printStackTrace // enforce dumping all exceptions to stderr
+		LOG.error("exception in " + AbstractProcessor.simpleName, th);
```
Comment (mmews-n4) [inline plugins/org.eclipse.n4js/src/org/eclipse/n4js/postprocessing/AbstractProcessor.xtend:219]: 👍

File: plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/build/XWorkspaceBuilder.java [hunk 2]
```diff
@@ -149,30 +150,20 @@ public BuildTask createReinitialBuildTask() {
 	}
 
 	/**
-	 * Perform a build on all projects
+	 * Run a full build on the entire workspace, i.e. build all projects.
 	 *
 	 * @param cancelIndicator
 	 *            cancellation support
-	 */
-	private IResourceDescription.Event doInitialBuild(CancelIndicator cancelIndicator) {
```
Comment (mmews-n4) [inline plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/build/XWorkspaceBuilder.java:157]: 👍

**Review by mmews-n4 — APPROVED** [review]
👍  several nice improvements!

**Comment by mor-n4** [comment]
Thank you! Builds green, merging ...

**Pull request merged**